nautobot_circuit_maintenance/models.py

```diff
@@ -240,37 +239,42 @@ class NotificationSource(OrganizationalModel):
     """Model for Notification Source configuration."""
 
     # Mark field as private so that it doesn't get included in ChangeLogging records!
-    _password = encrypt(models.CharField(max_length=100))
-    source_id = models.EmailField(max_length=100, unique=True, help_text="Email address used as identifier.")
-    url = models.CharField(max_length=200)
-    source_type = models.CharField(
-        default=NotificationSourceServerChoices.UNKNOWN,
-        max_length=50,
-        choices=NotificationSourceServerChoices,
-        null=True,
-        blank=True,
-        help_text="Type of source integration to retrieve the notifications.",
+    _password = encrypt(models.CharField(max_length=100, verbose_name="Password"))
+    source_id = models.EmailField(
+        max_length=100,
+        unique=True,
+        verbose_name="Source ID",
+        help_text="Identifier (i.e. email address) to use to authenticate.",
+    )
+    url = CustomURLField(
+        max_length=200,
```

Comment:
Might also need to increase this limit substantially once non-IMAP URLs are supported, but probably fine for the moment.